src/instance.rs

```diff
@@ -32,11 +34,15 @@ pub trait BoundObject<'py, T>: bound_object_sealed::Sealed {
 }
 
 mod bound_object_sealed {
-    pub trait Sealed {}
+    /// # Safety
+    ///
+    /// Type must be layout-compatible with `*mut ffi::PyObject`.
+    pub unsafe trait Sealed {}
```

Comment:
Yep I'm happy with the `unsafe` guarantee going here where users will never see it 😂